app/scripts/metamask-controller.js

```diff
@@ -2515,25 +2528,11 @@ export default class MetamaskController extends EventEmitter {
   setupControllerEventSubscriptions() {
     let lastSelectedAddress;
 
-    this.preferencesController.store.subscribe(async (state) => {
-      const { currentLocale } = state;
-
-      const { chainId } = this.networkController.state.providerConfig;
-      await updateCurrentLocale(currentLocale);
-
-      if (state.incomingTransactionsPreferences?.[chainId]) {
-        this.txController.startIncomingTransactionPolling();
-      } else {
-        this.txController.stopIncomingTransactionPolling();
-      }
-
-      // TODO: Remove once the preferences controller has been replaced with the core monorepo implementation
-      this.controllerMessenger.publish(
-        'PreferencesController:stateChange',
-        state,
-        [],
-      );
-    });
+    this.preferencesController.store.subscribe(
+      previousValueComparator((prevState, currState) => {
+        this.#onPreferencesControllerStateChange(currState, prevState);
```

Comment:
Can we be more explicit about `onPreferencesControllerStateChange` name? It seems rather generic and it is not the only place on the where we are subscribing to preferencesController state changes. Above in this file we are doing it as well for TokensRate and CurrencyRate controllers.
Another option would be move all that logic into this one method. But that should be outside of the scope for this PR.

Reply:
Previously we had many small preferences listeners spread around, so the name was intentional as the intent of this method was to create a generic container that would call other more specific methods.

So much of the logic of the `MetamaskController` is just within the constructor so I try to break it down into distinct methods whenever possible, for better readability and maintenance.